scylla/src/transport/session.rs

```diff
@@ -1653,7 +1653,10 @@ impl Session {
         QueryFut: Future<Output = Result<ResT, QueryError>>,
         ResT: AllowedRunQueryResTType,
     {
-        let mut last_error: Option<QueryError> = None;
+        // set default error as no known found as the query plan returns an empty iterator if there are no nodes in the plan
+        let mut last_error: Option<QueryError> = Some(QueryError::NoKnownNodeFoundError(
+            "Please confirm the supplied datacenters exists".to_string(),
+        ));
         let mut current_consistency: Consistency = context
```

Comment:
If I understand correctly, the problem happens when `query_plan` is empty. The question is, is this the only case when `last_error` can remain empty?
From brief reading of the code it looks like it, but I'm not 100% sure, and it could break in the future.
Could you rewrite this so that the new error is only returned when `query_plan` is empty, and otherwise the current behavior is preserved?